### ISSUE TITLE
gradient descent question

### ISSUE THREAD

**Issue by marcaureledivernois** [enesozeren/machine_learning_from_scratch#1]
Hello,
Thanks for your interesting post! 

Everything is clear to me, but I was wondering, mathematically speaking, how did you get the following expression? 

gradients["dA"+str(l-1)] = np.dot(parameters["W"+str(l)].T, gradients["dZ"+str(l)])

Best regards

**Comment by enesozeren** [comment]
Hello Marc,

Thanks for your question. Since you mentioned all other formulas were clear I can try to explain the intuition of the formula you ask here in a couple of sentences:

- We are calculating the gradients for the Loss function, so "dA" means derivative of Loss Function with respect to A.
- Since we are going backwards in layers during the backpropagation, when we want to calculate dA from layer "l-1" we should have already calculated dZ from layer "l".
- When you have dZ, dA for l-1 is basically distributing the gradient value of dZ with respect to the W (weights) in layer "l". Because Z for "l" = W for "l" * A from "l-1". When you apply the chain rule of derivation you have the dA for l-1 = W for l * dZ for l.

I hope I could make it clear for you :)

Enes

**Comment by marcaureledivernois** [comment]
Hello Enes,

thanks for your answer! It is clear now. I also found this blog that develops the network derivatives.

https://machinelearningmastery.com/application-of-differentiations-in-neural-networks/

Best regards
